hastexo/hastexo.py

```diff
@@ -678,6 +684,10 @@ def update_stack(self, data):
         in a transaction.
         """
         course_id, student_id = self.get_block_ids()
+        if not self.stack_name:
+            # Stack name can be occasionally end up being empty (after deleting
+            # learners state for example). If so, set it again.
+            self.stack_name = self.get_stack_name()
```

Comment:
Looks good. Can you think of a way to fix up a unit test so we get some coverage on this `if` block?

Reply:
Sure, I'll check that now.

Reply:
Added a test.